### ISSUE TITLE
keycloack roles

### ISSUE THREAD

**Issue by johnfg10** [FroMage/redpipe#15]
Hey, 
I have been testing out redpipe the last day or so and have set it up with keycloack is there any way to make it use keycloack roles with the RequiresPermissions annotation? 
Thanks,
John

**Comment by FroMage** [comment]
Yes, this application shows how to use keycloak roles with the annotation: https://github.com/FroMage/redpipe/tree/master/redpipe-example-wiki-keycloak-jooq

Did you follow this example?

http://redpipe.net/#authentication also has info. 

Also, my JWT for vertx 3.5.0 does not have the right line-info, so I assume you're running 3.5.1?

**Comment by johnfg10** [comment]
I followed http://redpipe.net/#authentication and yes I am running 3.5.1 is there anything else I need to do?

**Comment by FroMage** [comment]
So I tried with vertx 3.5.1 and it works here. This is with keycloack 3.3.0.CR2 and 3.3.0.Final. What version of keycloak did you try?

**Comment by FroMage** [comment]
Also, what does your `conf/config.json` look like?

**Comment by johnfg10** [comment]
i have keycloak 4.0.0.beta1(il try downgrading shortly it may be that) 
my config.json
```json
{
  "scan": ["io.github.johnfg10"],
  "keycloack":{
    "realm": "jtube",
    "auth-server-url": "http://localhost:8081/auth",
    "ssl-required": "external",
    "resource": "jtube",
    "credentials": {
      "secret": "3******f-8e*a-*c1f-a*02-****f3ab78**"
    },
    "use-resource-role-mappings": true,
    "confidential-port": 0,
    "policy-enforcer": {}
  }
}

**Comment by johnfg10** [comment]
sorry it took a few days ive been a little bit busy but ive downgraded to 3.4.3-final that also doesnt work

**Comment by FroMage** [comment]
So I have the following settings different:

Consent required: off
Direct Access Grants Enabled: on
Service Accounts Enabled: off
Authorization enabled: off
Valid redirect URIs: http://localhost:9000/callback/*

The rest are the same.

**Comment by johnfg10** [comment]
ive tried with the same settings and am still encountering the same errors
